lametro/static/js/wagtail_custom.js

```diff
@@ -0,0 +1,37 @@
+ALERT_COLORS = {
+	"primary": "#eb6864",
+	"secondary": "#aaa",
+	"success": "#22b24c",
+	"danger": "#369",
+	"warning": "#f5e625",
+	"info": "#f57a00",
+}
+
+function styleParent(e) {
+	if (e.target.classList.length > 0) {
+		e.target.classList = ""
+	}
+	e.target.classList.add(`alert-${e.target.value}`);
+}
+
+// Callback function to execute when mutations are observed
+const styleAlertTypeSelect = (mutationList, observer) => {
+	mutationList.map(mu => {
+		mu.addedNodes && mu.addedNodes.forEach(node => {
+			if (node.id === "id_type") {
+				styleParent({target: node})
+				node.addEventListener("change", styleParent)
+				observer.disconnect()
+				return
+			}
+		})
+	})
+};
+
+// Create an observer instance linked to the callback function
+const observer = new MutationObserver(styleAlertTypeSelect);
+
+const config = {childList: true, subtree: true};
+
+// Start observing the target node for configured mutations
+observer.observe(document.body, config);
```

Comment:
This whole rigamarole is necessary because Wagtail adds form fields to the DOM in JavaScript. In order to update the alert type select, we have to wait for it to show up, give it an initial style, then add an event listener for changes. The `MutationObserver` API is pretty dope: https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver

Reply:
h/t to @xmedr for this UI!

Reply:
Ah so is the mutation observer being used here mostly because we can't attach an event listener the usual way since the selector doesn't exist yet?